src/Castle.Windsor/Windsor/WindsorContainer.cs

```diff
@@ -586,6 +592,10 @@ public virtual object Resolve(String key, Type service, IDictionary arguments)
 		/// <returns></returns>
 		public virtual object Resolve(String key, Type service, object argumentsAsAnonymousType)
 		{
+			if (argumentsAsAnonymousType is IDictionary)
+			{
+				return kernel.Resolve(key, service, (IDictionary) argumentsAsAnonymousType);
+			}
```

Comment:
Any chance that `argumentsAsAnonymousType` could be `IReadOnlyDictionary<string, object>` here?

Reply:
If you cast `IReadOnlyDictionary<string, object>` to `object` then yes this overload will get called. These overloads love fighting with each other! :) Let me add another check.

Reply:
@fir3pho3nixx Another thought is that the ReadOnlyDictionary class also implements IDictionary, so you probably need ToDictionary there too. My main concern is that no matter what kind of dictionary I pass in, Castle Windows will never mutate it—even if it's *not* readonly.